Sources/Commands/SwiftTool.swift

```diff
@@ -497,39 +497,60 @@ public class SwiftTool {
 
     func getAuthorizationProvider() throws -> AuthorizationProvider? {
         var providers = [AuthorizationProvider]()
+        
         // netrc file has higher specificity than keychain so use it first
-        if let netrcConfigFile = try self.getNetrcConfigFile() {
-            providers.append(try NetrcAuthorizationProvider(path: netrcConfigFile, fileSystem: localFileSystem))
+        try providers.append(contentsOf: self.getNetrcAuthorizationProviders())
+
+#if canImport(Security)
+        if self.options.keychain {
+            providers.append(KeychainAuthorizationProvider(observabilityScope: self.observabilityScope))
         }
-        
-        // TODO: add --no-keychain option to allow opt-out
-//#if canImport(Security)
-//        providers.append(KeychainAuthorizationProvider(observabilityScope: self.observabilityScope))
-//#endif
+#endif
         
         return providers.isEmpty ? .none : CompositeAuthorizationProvider(providers, observabilityScope: self.observabilityScope)
     }
 
-    func getNetrcConfigFile() throws -> AbsolutePath? {
+    func getNetrcAuthorizationProviders() throws -> [NetrcAuthorizationProvider] {
```

Comment:
my expectation as a user would be a single netrc file loaded, with the following priority:
- custom one specified via cli flag
- project local
- home directory 

@yim-lee @mattt did you have a different semantic in mind?

Reply:
For `.netrc` files, I think that's a good option, yeah.

Alternatively, we could say:

- If you specify a `--netrc-path`, use only that
- Otherwise, consult project local (`./.netrc`) and fallback to the user shared (`~/.netrc`)

This approach would give us the convenience of cascading, while being able to keep the behavior you described (for example, if we only wanted to load the local or shared `.netrc` file, we could pass it as `--netrc-file`).

The only downside is that this is more complex, and lacking sufficient diagnostics, could make it difficult for the user to understand where credentials are coming from.

Reply:
looks like curl only support ~/.netrc: https://everything.curl.dev/usingcurl/netrc#the-netrc-file-format

Reply:
also note their comment about windows using `_netrc` instead of `.netrc`, should we do the same? cc @compnerd

Reply:
> I wonder if this is a good policy for us to due to security reasons - so users don't want to accidentally check in the `.netrc` file

FWIW, we added `.netrc` to the default project template `.gitignore` file with https://github.com/apple/swift-package-manager/pull/3511, which helps mitigate this.

Reply:
@yim-lee yes what you did is per spec, and correct. my question is more conceptual: if we got the spec right or if that would surprise users.

Reply:
@yim-lee @tomerd Since we have the spec from SE-0292 and this PR that implements the spec, I think we should go with that for now. If we find that this behavior is confusing or not what we want, we can amend this and update the spec / documentation in tandem.

Reply:
sgtm